docs/julia/Modules.md

```diff
@@ -0,0 +1,182 @@
+
+## Adding new modules
+1. Create a new folder `/src/julia/modules/YourModule/`.
+1. For every objective you want to support create a file `src/julia/modules/YourModule/YourModuleT.jl` with the following content
+    ```julia
+    module YourModuleT
+    # Adds modules in shared folder to LOAD_PATH
+    include("../../shared/load.jl")
+    using ADPerfTest
+    using TData
+    
+    export get_t_test
+
+    mutable struct YourModuleTContext
+        # context definition
+    end
+
+    function yourmodule_t_prepare!(context::YourModuleTContext, input::TInput)
+        # implementation
+    end
+
+    function yourmodule_t_calculate_objective!(context::YourModuleTContext, times)
+        for i in 1:times
+            # implementation
+        end
+    end
+
+    function yourmodule_t_calculate_jacobian!(context::YourModuleTContext, times)
+        for i in 1:times
+            # implementation
+        end
+    end
+
+    function yourmodule_t_output!(out::TOutput, context::YourModuleTContext)
+        # implementation
+    end
+
+    get_t_test() = Test{TInput, TOutput}(
+        YourModuleTContext(...),
+        yourmodule_t_prepare!,
+        yourmodule_t_calculate_objective!,
+        yourmodule_t_calculate_jacobian!,
+        yourmodule_t_output!
+    )
+
+    end
+    ```
+    Here `YourModule` is the name of your module in `PascalCase`, `yourmodule` - same in `all lowercase`, `T` is the capitalized short name of the objective (e.g. GMM, BA, one exception: `HAND` must be capitalized in the name of the file and the module, but written as `Hand` when it's a part of a name of a type, e.g. `HandInput`), and `t` is the short name of the objective in `all lowercase`. These are the expectations made by the [global runner](../GlobalRunner.md) and the plugin system implemented in [Julia runner](./Runner.md).
+
+1. <span id="itest-implementation">Define the `YourModuleTContext` type and implement the functions. An object of the `YourModuleTContext` type will be used by the [Julia runner](./Runner.md) to preserve the state of the benchmark between calls to these functions.
+    - ```julia
+      function yourmodule_t_prepare!(context::YourModuleTContext, input::TInput)
+      ```
+        Converts the input data from the `TInput` type in which it is provided by
+        the calling benchmark runner into the format optimized for use with the 
+        tested AD framework.
+        Stores it in the `context`.
+
+        Optionally, performs other preparatory activities need by the tested AD framework.
+    - ```julia 
+      function yourmodule_t_calculate_objective!(context::YourModuleTContext, times)
+      ``` 
+        Repeatedly computes the objective function `times` times for the input
+        stored in the context. Stores results in the context.
```

Comment:
Keeping a style, "context" should be inside \`\`, as I understand

Reply:
Fixed.